yugabyte/src/yugabyte/bank_improved.clj

```diff
@@ -0,0 +1,96 @@
+(ns yugabyte.bank-improved
+  "Simulates transfers between bank accounts with inserts and deletes"
+  (:refer-clojure :exclude
+                  [test])
+  (:require [clojure [pprint :refer [pprint]]]
+            [clojure.tools.logging :refer [info]]
+            [jepsen.tests.bank :as bank]
+            [knossos.op :as op]
+            [clojure.core.reducers :as r]
+            [jepsen
+             [generator :as gen]
+             [checker :as checker]
+             [store :as store]
+             [util :as util]]
+            [yugabyte.generator :as ygen]))
+
+(defn check-op
+  "Takes a single op and returns errors in its balance"
+  [accts total op]
+  (let [ks       (keys (:value op))
+        balances (vals (:value op))]
+    (cond
+      (some nil? balances)
+      {:type :nil-balance
+       :nils (->> (:value op)
+                  (remove val)
+                  (into {}))
+       :op   op}
+
+      (not= total (reduce + balances))
+      {:type  :wrong-total
+       :total (reduce + balances)
+       :op    op})))
+
+
+(defn err-badness
+  "Takes a bank error and returns a number, depending on its type. Bigger
+  numbers mean more egregious errors."
+  [test err]
+  (case (:type err)
+    :nil-balance    (count (:nils err))
+    :wrong-total    (Math/abs
+                      (float
+                        (/ (- (:total err) (:total-amount test))
+                           (:total-amount test))))))
+
+(defn checker
```

Comment:
Not sure that negative check is necessary here (As far as I understand it's disabled by default now)
Because it's a custom bank workload it work with crunches right now, I understand it. 

Problem is that Jepsen operates only in area of initial keys. So update will be called only between (1, END_KEY) and at some point all transactions will fail because there is no balance on accounts.